### PR TITLE
EvmBuilder and its dependencies #1

### PR DESCRIPTION
**Pull request by x-senpai-x**
FIles Overview : 
Common File Common.go
1)EvmBuilder.go: generic builder for creating EVM instances. creating a default builder, updating the database, and modifying the environment settings for the EVM.
2)Handler.go:
This file defines the Handler and EvmHandler types for managing Ethereum Virtual Machine (EVM) execution contexts in Go. It includes configuration structures (HandlerCfg), methods for initializing handlers for different Ethereum specifications (mainnet and optimism) 
Build FLags used for optimism 
to enable them add it in gopls through settings
"gopls": {
    "buildFlags": ["optimism"]
}

3)Context.go : creating and managing the contexts necessary for executing EVM operations
4)EvmContext.go:managing the EVM context within a specified database environment
5)Precompiles.go: integrating and executing precompiles in an EVM environment

6)InnerEvmContext.go:manage the internal state of the EVM execution environment. L1 block info holds parameters related to layer 1 block information
7)Journal.go:managing the state of evm environment
8)Journal_Entry.go :tracking state changes (account changes and storage modifications)

Files contain some commented out functions  And some functions with comment not in use No need to test them

### CODE REVIEW

**Comment by x-senpai-x** [comment]
Marshal Unmarshal might need to be implemented again 
I am not very confident about their implementation

File: execution/evm/handler.go [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+func (p PlainRegister[H, EXT, DB]) Register(handler *EvmHandler[H, EXT, DB]) {
+	p.RegisterFn(handler)
+}
+
+// Register method for BoxRegister, to satisfy HandleRegisters interface.
+func (b BoxRegister[H, EXT, DB]) Register(handler *EvmHandler[H, EXT, DB]) {
+	b.RegisterFn(handler)
+}
+/*
+func (s SpecId) IsEnabledIn(spec SpecId) bool {
```
Comment (gerceboss) [inline execution/evm/handler.go:115]: @x-senpai-x ,remove all these unnecessary comments from this PR

**Comment by gerceboss** [comment]
@x-senpai-x , please donot add any undefined things in a PR , you can always add them in the next ones like the ones you made in a series, if the build and tests fail , then I can't make sure if the integrated code at the end will work.

**Comment by x-senpai-x** [comment]
@gerceboss The issue is that in evm module most of the things are related to each other 
So if I ensure that no undefined thing remains then I'll end up having all 27 files in the same pr 
> @x-senpai-x , please donot add any undefined things in a PR , you can always add them in the next ones like the ones you made in a series, if the build and tests fail , then I can't make sure if the integrated code at the end will work.
>

File: execution/evm/handler.go (hunk 1 above)
Comment (gerceboss) [reply execution/evm/handler.go:115]: Can you please do the suggested changes? And also mention the sequence of PRs for the evm.go

**Pull request closed without merging**